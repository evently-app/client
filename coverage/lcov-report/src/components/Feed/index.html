<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Feed</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/Feed
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">40.54% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>45/111</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>29/58</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">23.26% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>10/43</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40.19% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>43/107</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Filter.js"><a href="Filter.js.html">Filter.js</a></td>
	<td data-value="47.46" class="pic low"><div class="chart"><div class="cover-fill" style="width: 47%;"></div><div class="cover-empty" style="width:53%;"></div></div></td>
	<td data-value="47.46" class="pct low">47.46%</td>
	<td data-value="59" class="abs low">28/59</td>
	<td data-value="61.11" class="pct medium">61.11%</td>
	<td data-value="36" class="abs medium">22/36</td>
	<td data-value="29.17" class="pct low">29.17%</td>
	<td data-value="24" class="abs low">7/24</td>
	<td data-value="47.27" class="pct low">47.27%</td>
	<td data-value="55" class="abs low">26/55</td>
	</tr>

<tr>
	<td class="file low" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="32.69" class="pic low"><div class="chart"><div class="cover-fill" style="width: 32%;"></div><div class="cover-empty" style="width:68%;"></div></div></td>
	<td data-value="32.69" class="pct low">32.69%</td>
	<td data-value="52" class="abs low">17/52</td>
	<td data-value="31.82" class="pct low">31.82%</td>
	<td data-value="22" class="abs low">7/22</td>
	<td data-value="15.79" class="pct low">15.79%</td>
	<td data-value="19" class="abs low">3/19</td>
	<td data-value="32.69" class="pct low">32.69%</td>
	<td data-value="52" class="abs low">17/52</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Apr 14 2019 20:53:00 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
