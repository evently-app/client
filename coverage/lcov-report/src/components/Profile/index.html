<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Profile</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/Profile
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">40.35% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>23/57</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20.83% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/24</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">29.41% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/17</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40.74% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>22/54</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Dial.js"><a href="Dial.js.html">Dial.js</a></td>
	<td data-value="29.55" class="pic low"><div class="chart"><div class="cover-fill" style="width: 29%;"></div><div class="cover-empty" style="width:71%;"></div></div></td>
	<td data-value="29.55" class="pct low">29.55%</td>
	<td data-value="44" class="abs low">13/44</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="18" class="abs low">2/18</td>
	<td data-value="30.77" class="pct low">30.77%</td>
	<td data-value="13" class="abs low">4/13</td>
	<td data-value="30.95" class="pct low">30.95%</td>
	<td data-value="42" class="abs low">13/42</td>
	</tr>

<tr>
	<td class="file medium" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="76.92" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 76%;"></div><div class="cover-empty" style="width:24%;"></div></div></td>
	<td data-value="76.92" class="pct medium">76.92%</td>
	<td data-value="13" class="abs medium">10/13</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="6" class="abs medium">3/6</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="12" class="abs medium">9/12</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Apr 14 2019 20:53:00 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
