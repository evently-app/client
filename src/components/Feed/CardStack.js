// import our swipeable card component, handle fetching of more data when the user swipes a certain number of cards