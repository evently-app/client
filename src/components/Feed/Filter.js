import React, { PureComponent } from "react";
import { Animated, View, TouchableOpacity, StyleSheet } from "react-native";
import { connect } from "react-redux";
import Interactable from "react-native-interactable";
import Haptics from "react-native-haptic-feedback";

import {
	BeginTransition,
	EndTransition,
	SnapOpen,
	SnapClosed,
	ScrollTimeSelection,
	ScrollTypeSelection
} from "../../redux/filter";
import { Header, SubHeader, Paragraph } from "../universal/Text";
import {
	SB_HEIGHT,
	SCREEN_WIDTH,
	SCREEN_HEIGHT,
	IS_X,
	CATEGORIES,
	TIME_TYPES
} from "../../lib/constants";
import { colors } from "../../lib/styles";

const CLOSED_POINT = { y: 0 };
const OPEN_POINT = { y: 150 };
const BOUNDARIES = {
	top: -1,
	bottom: 225,
	haptics: true
};

const FILTER_DRAG_RANGE = [0, 50, 150];
const TYPE_SNAP_POINTS = CATEGORIES.map((category, i) => ({ x: ((9 - 2 * i) * SCREEN_WIDTH) / 8 }));
const TIME_SNAP_POINTS = TIME_TYPES.map((time, i) => ({ x: -1 * (((i - 1) * SCREEN_WIDTH) / 3) }));

class Filter extends PureComponent {
	state = {
		time: 0,
		type: 0
	};

	timeXOffset = new Animated.Value(0);
	typeXOffset = new Animated.Value(0);

	handleOnDrag = ({ nativeEvent }) => {
		const { state } = nativeEvent;
		const { open, onPress, BeginTransition, EndTransition } = this.props;

		if (state === "start") BeginTransition();
		// else EndTransition();
	};

	handleOnSnap = ({ nativeEvent }) => {
		const { index } = nativeEvent;
		const {
			SnapOpen,
			SnapClosed,
			timeSelection,
			typeSelection,
			ScrollTimeSelection,
			ScrollTypeSelection
		} = this.props;

		const { time, type } = this.state;

		Haptics.trigger("impactLight");

		// weird necessary fix
		this.timeXOffset.setValue(-1 * (((timeSelection - 1) * SCREEN_WIDTH) / 3));
		this.typeXOffset.setValue(((9 - typeSelection * 2) * SCREEN_WIDTH) / 8);

		// snap closed, update the selected time and type filters
		if (index == 0) SnapClosed({ time, type });
		else SnapOpen();
	};

	// update time selection in redux
	handleTimeScroll = ({ nativeEvent }) => {
		const { index } = nativeEvent;

		// update local filter state with new selection
		Haptics.trigger("impactLight");
		this.setState({ time: index });
	};

	// update type selection in redux
	handleTypeScroll = ({ nativeEvent }) => {
		const { index } = nativeEvent;

		// update local filter state with new selection
		Haptics.trigger("impactLight");
		this.setState({ type: index });
	};

	timeSelectionStyle = index => {
		const { open, transitioning, filterDrag } = this.props;
		const { time } = this.state;

		return {
			// transform: [
			// 	{
			// 		scale: this.timeXOffset.interpolate({
			// 			inputRange: [0, SCREEN_WIDTH / 3, (2 * SCREEN_WIDTH) / 3],
			// 			outputRange: [index === 0 ? 1 : 0.7, index === 1 ? 1 : 0.7, index === 2 ? 1 : 0.7]
			// 		})
			// 	}
			// ],
			...styles.timeSelection,
			opacity:
				!transitioning && open
					? this.timeXOffset.interpolate({
							inputRange: TIME_SNAP_POINTS.map(({ x }) => x).reverse(),
							outputRange: TIME_SNAP_POINTS.map((point, i) => (index === i ? 1 : 0.5)).reverse(),
							extrapolate: "clamp"
					  })
					: filterDrag.interpolate({
							inputRange: FILTER_DRAG_RANGE,
							outputRange: [
								index === time ? 1 : 0,
								index === time ? 1 : 0.1,
								index === time ? 1 : 0.3
							],
							extrapolate: "clamp"
					  })
		};
	};

	typeSelectionStyle = index => {
		const { open, transitioning, filterDrag } = this.props;
		const { type } = this.state;

		return {
			...styles.typeSelection,
			opacity:
				!transitioning && open
					? this.typeXOffset.interpolate({
							inputRange: TYPE_SNAP_POINTS.map(({ x }) => x).reverse(),
							outputRange: TYPE_SNAP_POINTS.map((point, i) => (index === i ? 1 : 0.5)).reverse(),
							extrapolate: "clamp"
					  })
					: filterDrag.interpolate({
							inputRange: FILTER_DRAG_RANGE,
							outputRange: [
								index === type ? 1 : 0,
								index === type ? 1 : 0.25,
								index === type ? 1 : 0.3
							],
							extrapolate: "clamp"
					  })
		};
	};

	openFilter = () => {
		this.filter.snapTo({ index: 1 });
	};

	closeFilter = () => {
		this.filter.snapTo({ index: 0 });
	};

	render() {
		const { open, filterDrag } = this.props;
		const { type } = this.state;

		const animatedLocation = {
			transform: [
				{
					translateY: filterDrag.interpolate({
						inputRange: FILTER_DRAG_RANGE,
						outputRange: [0, -35, -105]
					})
				}
			]
		};

		const animatedTime = {
			transform: [
				{
					translateY: filterDrag.interpolate({
						inputRange: FILTER_DRAG_RANGE,
						outputRange: [-17, -37, -93]
					})
				}
			]
		};

		const animatedType = {
			transform: [
				{
					translateY: filterDrag.interpolate({
						inputRange: FILTER_DRAG_RANGE,
						outputRange: [-30, -65, -115]
					})
				}
			]
		};

		const animatedOpacity = {
			opacity: filterDrag.interpolate({
				inputRange: FILTER_DRAG_RANGE,
				outputRange: [0, 0.25, 0.5],
				extrapolate: "clamp"
			})
		};

		const animatedOpacity2 = {
			opacity: filterDrag.interpolate({
				inputRange: FILTER_DRAG_RANGE,
				outputRange: [0, 0.5, 1],
				extrapolate: "clamp"
			})
		};

		const indicatorOpacity = {
			opacity: filterDrag.interpolate({
				inputRange: FILTER_DRAG_RANGE,
				outputRange: [1, 0.2, 0],
				extrapolate: "clamp"
			})
		};

		return (
			<>
				<Interactable.View
					verticalOnly
					animatedNativeDriver
					snapPoints={[CLOSED_POINT, OPEN_POINT]}
					ref={Interactable => (this.filter = Interactable)}
					onDrag={this.handleOnDrag}
					onSnapStart={this.handleOnSnap}
					boundaries={BOUNDARIES}
					initialPosition={CLOSED_POINT}
					style={styles.filterContainer}
					animatedValueY={filterDrag}
				>
					<Paragraph animated style={{ ...animatedLocation, ...animatedOpacity }}>
						I want events
					</Paragraph>
					<Header animated style={animatedLocation}>
						Nearby
					</Header>
					<Paragraph animated style={{ ...animatedTime, ...animatedOpacity }}>
						that are
					</Paragraph>
					<Interactable.View
						horizontalOnly
						animatedNativeDriver
						ref={Interactable => (this.timeSelector = Interactable)}
						dragEnabled={open}
						snapPoints={TIME_SNAP_POINTS}
						initialPosition={TIME_SNAP_POINTS[0]}
						onSnapStart={this.handleTimeScroll}
						style={[animatedTime, styles.horizontalSelector]}
						animatedValueX={this.timeXOffset}
					>
						{TIME_TYPES.map(({ title }, i) => (
							<TouchableOpacity key={i} onPress={() => this.timeSelector.snapTo({ index: i })}>
								<SubHeader animated style={this.timeSelectionStyle(i)}>
									{title}
								</SubHeader>
							</TouchableOpacity>
						))}
					</Interactable.View>
					<Paragraph animated style={{ ...animatedType, ...animatedOpacity }}>
						and
					</Paragraph>
					<Interactable.View
						horizontalOnly
						animatedNativeDriver
						ref={Interactable => (this.typeSelector = Interactable)}
						dragEnabled={open}
						snapPoints={TYPE_SNAP_POINTS}
						initialPosition={TYPE_SNAP_POINTS[0]}
						onSnapStart={this.handleTypeScroll}
						style={[
							animatedType,
							animatedOpacity2,
							styles.horizontalSelector,
							{ width: 2.5 * SCREEN_WIDTH }
						]}
						animatedValueX={this.typeXOffset}
					>
						{CATEGORIES.map(({ title }, i) => (
							<TouchableOpacity key={i} onPress={() => this.typeSelector.snapTo({ index: i })}>
								<SubHeader animated style={this.typeSelectionStyle(i)}>
									{title}
								</SubHeader>
							</TouchableOpacity>
						))}
					</Interactable.View>
				</Interactable.View>
				{type !== 0 && (
					<Animated.View style={[styles.typeIndicator, indicatorOpacity]}>
						<Paragraph>{CATEGORIES[typeSelection].title}</Paragraph>
					</Animated.View>
				)}
				{open && (
					<TouchableOpacity
						activeOpacity={1}
						style={styles.closeFilterButton}
						onPressIn={this.closeFilter}
					/>
				)}
				<TouchableOpacity
					style={styles.toggleButton}
					onPress={open ? this.closeFilter : this.openFilter}
				/>
			</>
		);
	}
}

const styles = StyleSheet.create({
	center: {
		alignItems: "center",
		justifyContent: "center"
	},
	filterContainer: {
		position: "absolute",
		top: SB_HEIGHT + (IS_X ? 0 : 10) - 250,
		left: 0,
		right: 0,
		height: 400,
		alignItems: "center",
		justifyContent: "flex-end"
	},
	horizontalSelector: {
		width: SCREEN_WIDTH,
		paddingBottom: 25,
		flexDirection: "row",
		justifyContent: "space-around",
		alignItems: "center"
	},
	timeSelection: {
		width: SCREEN_WIDTH / 3,
		textAlign: "center"
	},
	typeSelection: {
		width: SCREEN_WIDTH / 4,
		textAlign: "center"
	},
	typeIndicator: {
		position: "absolute",
		paddingHorizontal: 12,
		paddingVertical: 6,
		borderRadius: 10,
		backgroundColor: colors.lightpurple,
		top: SB_HEIGHT + 12,
		left: 30
	},
	toggleButton: {
		position: "absolute",
		top: SB_HEIGHT - 10,
		left: 0,
		right: 0,
		height: 80
	},
	closeFilterButton: {
		position: "absolute",
		top: 250,
		bottom: 0,
		left: 0,
		right: 0
	}
});

const mapStateToProps = ({ filter }) => {
	const { open, transitioning, timeSelection, typeSelection } = filter;
	return {
		open,
		transitioning,
		timeSelection,
		typeSelection
	};
};

const mapDispatchToProps = {
	BeginTransition,
	EndTransition,
	SnapOpen,
	SnapClosed,
	ScrollTimeSelection,
	ScrollTypeSelection
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(Filter);
