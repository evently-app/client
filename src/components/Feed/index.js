import React, { Component, Fragment } from "react";
import { Animated, Easing, View, Text, TouchableOpacity, StyleSheet } from "react-native";
import { connect } from "react-redux";
import _ from "lodash";

import Spinner from "../universal/Spinner";
import Filter from "./Filter";
import EventCard from "../EventCard";
import Swipeable from "../EventCard/Swipeable";

import { SCREEN_WIDTH, SCREEN_HEIGHT, SB_HEIGHT } from "../../lib/constants";
import { LoadQueue, UpdateQueue } from "../../redux/queue";
import { SwipeRight, SwipeLeft } from "../../redux/timeline";

class Feed extends Component {
	state = {
		animatedValues: {}
	};

	animatedEntry = new Animated.Value(1);
	filterDrag = new Animated.Value(0);

	componentDidMount() {
		const { LoadQueue, filterTime, filterType } = this.props;
		LoadQueue({ filterTime, filterType }).then(this.entryAnimation);
	}

	static getDerivedStateFromProps(props, state) {
		const {
			UpdateQueue,
			queue,
			currentTypeFilter,
			currentTimeFilter,
			selectedType,
			selectedTime
		} = props;
		const { animatedValues } = state;

		// if there's a mismatch then new events have been loaded
		if (queue.length !== _.size(state.animatedValues)) {
			let newAnimatedValues = {};
			queue.forEach(({ id }) => {
				if (animatedValues[id] == undefined) newAnimatedValues[id] = new Animated.Value(0);
			});

			return { animatedValues: { ...newAnimatedValues, ...animatedValues } };
		}

		// if the filter has changed hide the cards and update the queue
		if (currentTypeFilter !== selectedType || currentTimeFilter != selectedTime) {
			console.log("updating queue");

			this.exitAnimation();
			UpdateQueue({ filterTime: selectedTime, filterType: selectedType }).then(this.entryAnimation);

			return { animatedValues: {} };
		}

		return null;
	}

	entryAnimation = () => {
		Animated.timing(this.animatedEntry, {
			toValue: 0,
			duration: 300,
			easing: Easing.quad,
			useNativeDriver: true
		}).start();
	};

	exitAnimation = () => {
		Animated.timing(this.animatedEntry, {
			toValue: 1,
			duration: 300,
			easing: Easing.quad,
			useNativeDriver: true
		}).start();
	};

	popCard = ({ id }) => {
		const { UpdateQueue, filterTime, filterType } = this.props;
		const { animatedValues } = this.state;

		// if fewer than five events remaining in the queue, reload
		if (_.size(animatedValues) < 5) UpdateQueue({ filterTime, filterType });

		this.setState({
			animatedValues: _.omit(animatedValues, id)
		});
	};

	onSwipeCardRight = card => {
		this.props.SwipeRight(card);
		this.popCard(card);
	};

	onSwipeCardLeft = card => {
		this.props.SwipeLeft(card);
		this.popCard(card);
	};

	handleOnStartSwipe = () => {
		// this.fetchCards();
	};

	render() {
		const { animatedValues } = this.state;
		const { queue, loading, userLocation } = this.props;

		const cardContainerStyle = {
			opacity: this.animatedEntry.interpolate({
				inputRange: [0, 1],
				outputRange: [1, 0]
			}),
			transform: [
				{
					translateY: this.filterDrag.interpolate({
						inputRange: [0, 100],
						outputRange: [0, 100]
					})
				},
				{
					scale: this.animatedEntry.interpolate({
						inputRange: [0, 1],
						outputRange: [1, 0.9]
					})
				}
			]
		};

		const first = queue.length - 1;
		const cards = (
			<Animated.View style={[styles.center, cardContainerStyle]}>
				{queue.map((card, i) => (
					<Swipeable
						key={card.id}
						id={card.id}
						index={queue.length - i}
						swipeAmount={animatedValues[card.id]}
						scaleAmount={i !== first ? animatedValues[queue[i + 1].id] : null}
						onStartSwipe={this.handleOnStartSwipe}
						onSwipeRight={() => this.onSwipeCardRight(card)}
						onSwipeLeft={() => this.onSwipeCardLeft(card)}
					>
						<EventCard userLocation={userLocation} {...card} />
					</Swipeable>
				))}
			</Animated.View>
		);

		return (
			<View style={styles.container}>
				{loading ? <Spinner /> : cards}
				<Filter filterDrag={this.filterDrag} />
			</View>
		);
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		width: SCREEN_WIDTH,
		height: SCREEN_HEIGHT,
		overflow: "hidden",
		alignItems: "center",
		justifyContent: "center"
	},
	center: {
		alignItems: "center",
		justifyContent: "center"
	}
});

const mapStateToProps = ({ queue, filter, user }) => {
	return {
		queue: queue.queue,
		currentTypeFilter: queue.currentTypeFilter,
		currentTimeFilter: queue.currentTimeFilter,
		loading: queue.isLoadingQueue,
		selectedTime: filter.selectedTime,
		selectedType: filter.selectedType,
		userLocation: user.location
	};
};

const mapDispatchToProps = {
	LoadQueue,
	UpdateQueue,
	SwipeRight,
	SwipeLeft
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(Feed);
