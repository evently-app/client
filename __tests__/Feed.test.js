import "react-native";
import React from "react";
import { Feed } from "../src/components/Feed";
import { Filter } from "../src/components/Feed/Filter";

jest.mock("@mapbox/react-native-mapbox-gl");
jest.mock("react-native-permissions");

const Animated = {
	Value: class {
		constructor(initialValue) {
			this._value = initialValue;
		}

		interpolate() {}
	}
};

it("Feed renders correctly", () => {
	const snap = shallow(<Feed queue={[]} />);
	expect(snap).toMatchSnapshot();
});

it("Filters renders correctly", () => {
	const snap = shallow(<Filter filterDrag={new Animated.Value(0)} />);
	expect(snap).toMatchSnapshot();
});
