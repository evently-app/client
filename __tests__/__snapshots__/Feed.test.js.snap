// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": 1334,
      "justifyContent": "center",
      "overflow": "hidden",
      "width": 750,
    }
  }
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
            Object {
              "scale": 0.9,
            },
          ],
        },
      ]
    }
  />
  <Connect(Filter)
    filterDrag={0}
  />
</View>
`;

exports[`Filters renders correctly 1`] = `
<Fragment>
  <WrappedAnimatedInteractableView
    animatedNativeDriver={true}
    animatedValueY={
      Value {
        "_value": 0,
      }
    }
    boundaries={
      Object {
        "bottom": 225,
        "haptics": true,
        "top": -1,
      }
    }
    initialPosition={
      Object {
        "y": 0,
      }
    }
    onDrag={[Function]}
    onSnapStart={[Function]}
    snapPoints={
      Array [
        Object {
          "y": 0,
        },
        Object {
          "y": 150,
        },
      ]
    }
    style={
      Object {
        "alignItems": "center",
        "height": 400,
        "justifyContent": "flex-end",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": -220,
      }
    }
    verticalOnly={true}
  >
    <Paragraph
      animated={true}
      style={
        Object {
          "opacity": undefined,
          "transform": Array [
            Object {
              "translateY": undefined,
            },
          ],
        }
      }
    >
      I want events
    </Paragraph>
    <Header
      animated={true}
      style={
        Object {
          "transform": Array [
            Object {
              "translateY": undefined,
            },
          ],
        }
      }
    >
      Nearby
    </Header>
    <Paragraph
      animated={true}
      style={
        Object {
          "opacity": undefined,
          "transform": Array [
            Object {
              "translateY": undefined,
            },
          ],
        }
      }
    >
      that are
    </Paragraph>
    <WrappedAnimatedInteractableView
      animatedNativeDriver={true}
      animatedValueX={0}
      horizontalOnly={true}
      initialPosition={
        Object {
          "x": 250,
        }
      }
      onSnapStart={[Function]}
      snapPoints={
        Array [
          Object {
            "x": 250,
          },
          Object {
            "x": -0,
          },
          Object {
            "x": -250,
          },
        ]
      }
      style={
        Array [
          Object {
            "transform": Array [
              Object {
                "translateY": undefined,
              },
            ],
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-around",
            "paddingBottom": 25,
            "width": 750,
          },
        ]
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        key="0"
        onPress={[Function]}
      >
        <SubHeader
          animated={true}
          style={
            Object {
              "opacity": undefined,
              "textAlign": "center",
              "width": 250,
            }
          }
        >
          Upcoming
        </SubHeader>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        key="1"
        onPress={[Function]}
      >
        <SubHeader
          animated={true}
          style={
            Object {
              "opacity": undefined,
              "textAlign": "center",
              "width": 250,
            }
          }
        >
          Next Week
        </SubHeader>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        key="2"
        onPress={[Function]}
      >
        <SubHeader
          animated={true}
          style={
            Object {
              "opacity": undefined,
              "textAlign": "center",
              "width": 250,
            }
          }
        >
          This Month
        </SubHeader>
      </TouchableOpacity>
    </WrappedAnimatedInteractableView>
    <Paragraph
      animated={true}
      style={
        Object {
          "opacity": undefined,
          "transform": Array [
            Object {
              "translateY": undefined,
            },
          ],
        }
      }
    >
      and
    </Paragraph>
    <WrappedAnimatedInteractableView
      animatedNativeDriver={true}
      animatedValueX={0}
      horizontalOnly={true}
      initialPosition={
        Object {
          "x": 656.25,
        }
      }
      onSnapStart={[Function]}
      snapPoints={
        Array [
          Object {
            "x": 656.25,
          },
          Object {
            "x": 468.75,
          },
          Object {
            "x": 281.25,
          },
          Object {
            "x": 93.75,
          },
          Object {
            "x": -93.75,
          },
          Object {
            "x": -281.25,
          },
          Object {
            "x": -468.75,
          },
          Object {
            "x": -656.25,
          },
        ]
      }
      style={
        Array [
          Object {
            "transform": Array [
              Object {
                "translateY": undefined,
              },
            ],
          },
          Object {
            "opacity": undefined,
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-around",
            "paddingBottom": 25,
            "width": 750,
          },
          Object {
            "width": 1500,
          },
        ]
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        key="0"
        onPress={[Function]}
      >
        <SubHeader
          animated={true}
          style={
            Object {
              "opacity": undefined,
              "textAlign": "center",
              "width": 187.5,
            }
          }
        >
          Anything
        </SubHeader>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        key="1"
        onPress={[Function]}
      >
        <SubHeader
          animated={true}
          style={
            Object {
              "opacity": undefined,
              "textAlign": "center",
              "width": 187.5,
            }
          }
        >
          Free
        </SubHeader>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        key="2"
        onPress={[Function]}
      >
        <SubHeader
          animated={true}
          style={
            Object {
              "opacity": undefined,
              "textAlign": "center",
              "width": 187.5,
            }
          }
        >
          Shows
        </SubHeader>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        key="3"
        onPress={[Function]}
      >
        <SubHeader
          animated={true}
          style={
            Object {
              "opacity": undefined,
              "textAlign": "center",
              "width": 187.5,
            }
          }
        >
          Sports
        </SubHeader>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        key="4"
        onPress={[Function]}
      >
        <SubHeader
          animated={true}
          style={
            Object {
              "opacity": undefined,
              "textAlign": "center",
              "width": 187.5,
            }
          }
        >
          Art
        </SubHeader>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        key="5"
        onPress={[Function]}
      >
        <SubHeader
          animated={true}
          style={
            Object {
              "opacity": undefined,
              "textAlign": "center",
              "width": 187.5,
            }
          }
        >
          Nightlife
        </SubHeader>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        key="6"
        onPress={[Function]}
      >
        <SubHeader
          animated={true}
          style={
            Object {
              "opacity": undefined,
              "textAlign": "center",
              "width": 187.5,
            }
          }
        >
          Family
        </SubHeader>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        key="7"
        onPress={[Function]}
      >
        <SubHeader
          animated={true}
          style={
            Object {
              "opacity": undefined,
              "textAlign": "center",
              "width": 187.5,
            }
          }
        >
          Professional
        </SubHeader>
      </TouchableOpacity>
    </WrappedAnimatedInteractableView>
  </WrappedAnimatedInteractableView>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "height": 80,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 10,
      }
    }
  />
</Fragment>
`;
